### ISSUE TITLE
How to retrieve encryption key?

### ISSUE THREAD

**Issue by ronoliverclarin** [brantje/passman#120]
I forgot my encryption key and I tried to use a new one. But, It gives me blank (cannot see my large number of passwords.

Now, I am getting helpless. How can I retrieve my encryption key?

**Comment by tiker** [comment]
As far as I know, you can not retrieve your encryption key(s).  I have not seen anything in the code that would indicate that they are stored or retrievable (yet).

A different key (a new key) gives you a new clean password space to work with.  You can start over with a new key but retrieval of an old key isn't possible (unless one of the main developers more familiar with the software says otherwise).

**Comment by brantje** [comment]
Retrieving the key is impossible,  everything is encrypted client side, and the key is never send to the server. It would be possible to write a brute force script. Won't be to hard to do that.
That you get an empty list is because you entered a wrong password, there for passman creates a new vault. This 'system' will be removed soon, as it creates confusion for a lot of people. 
Let me know if you want a simple brute force tool, can't make any promise when it could be ready.

**Comment by tiker** [comment]
Removing the vaults?  That's not good... Is there something to replace the vaults to keep the same functionality?  This is one of the reasons I started using PassMan.

**Comment by brantje** [comment]
I have to discuss that with @animalillo he said something about only 1 vault. But I will add an message that a new vault is created when you entered a other password.

**Comment by tiker** [comment]
Ok.  Hopefully they're not going away or implement some sort of "Advanced User" interface that keeps the existing multiple vaults.  I need the different encryption key vaults for what I do and store.

If it is planned to eliminate the different vaults, please let me (and others) know in advance.

Thanks.

**Comment by animalillo** [comment]
I don't think we will remove vaults, but completely recode the way they work.
Recovering the password via bruteforce can take ages, as the system is using AES to encrypt the passwords. If you knew a part of the password, the length and other details, the script could be improved, but the main idea behind passman, or any other password manager it's to keep your passwords away from unwanted eyes.

PS: Maybe add a disclaimer on the wellcome screen saying that there is no way to recover forgotten passwords?

**Comment by Timmeey** [comment]
I have kind of the same problem. Unable to remember the correct password. But i managed to compile a liste with 4k passwords, which should contain the correct password. Is there an easy way to check all those passwords (I have access to the passman Database to retriev data)? Maybe using "John the ripper"?

**Comment by animalillo** [comment]
you could easily make a javascript dictionary attack over one of your cyphered passwords, and then use your found password to access passman and change the password to one you can remember ;P
